### PR TITLE
Correctly handle upper alpha to h2 full line bolds

### PR DESCRIPTION
**Pull request by jlward**
If a list get converted to an h2 (upper alpha in root) and is fully bold, the bold does not get stripped.

### CODE REVIEW

File: docx2html/core.py [hunk 1]
```diff
@@ -1075,14 +1075,14 @@ def get_p_data(p, meta_data, is_td=False):
...
-        # Check to see if the whole line is bold or italics.
-        whole_line_bold, whole_line_italics = whole_line_styled(p)
-        p_is_header = bool(is_header(p, meta_data) and not is_li(p, meta_data))
 
-        # Only remove bold or italics if this tag is an h tag.
-        remove_bold = p_is_header and whole_line_bold
-        remove_italics = p_is_header and whole_line_italics
+    if (
+            not is_td and
+            is_header(p, meta_data) or
```
Comment (winhamwr) [inline docx2html/core.py]: This kind of checking is going to lead to some really ugly code as we continue to add more cases. At some point, the architecture here will have to improve to accommodate a list of functions that might alter the output.

File: docx2html/core.py [hunk 2]
```diff
@@ -1075,14 +1075,14 @@ def get_p_data(p, meta_data, is_td=False):
...
+            _is_top_level_upper_roman(p, meta_data)
```
Comment (winhamwr) [inline docx2html/core.py]: Could use some comments here (similar to those which were removed). I would also split this up in to two booleans that are or'd together. `converting_to_header` and `is_non_td_header` or something

File: docx2html/tests/test_xml.py [hunk 3]
```diff
@@ -32,19 +32,27 @@ def _create_t_tag(text):
...
 
 
-def _create_li(text, ilvl, numId):
+def _create_li(text, ilvl, numId, bold=False):
     text = _create_t_tag(text)
     return DOCUMENT_LI_TEMPLATE % {
         'text': text,
         'ilvl': ilvl,
         'numId': numId,
+        'bold': _bold(bold),
```
Comment (winhamwr) [inline docx2html/tests/test_xml.py]: At least use the named parameter here. This is pretty weird-looking code, though. `_bold(bold)`

File: docx2html/core.py (hunk 2 above)
Comment (jlward) [inline docx2html/core.py]: I don't think splitting this into two booleans will help. `convert_to_header` is by definition `is_non_td_header`

File: docx2html/tests/__init__.py [hunk 4]
```diff
@@ -39,7 +41,9 @@
             </w:numPr>
         </w:pPr>
         <w:r>
-            <w:rPr/>
+            <w:rPr>
+                %(bold)s
```
Comment (winhamwr) [inline docx2html/tests/__init__.py:45]: Looks like you're to the point where using something like Jinja2 to render your XML would make this look way less weird. Having a `%(bold)s` that has the value `<w:b/>` is getting difficult to read.

Comment (jlward) [inline docx2html/tests/__init__.py:45]: The problem with that is that if its not bold we don't want anything there

Comment (jlward) [inline docx2html/tests/__init__.py:45]: #6 will address this

**Pull request merged**